File: hex_to_ascii.go
package main

import (
    "fmt"
    "encoding/hex"
)

func main() {
    // Get the hex string from the user.
    fmt.Print("Enter a hex string: ")
    var hex string
    fmt.Scanln(&hex)

    // Convert the hex string to ASCII.
    ascii, err := hex.DecodeString(hex)
    if err != nil {
        fmt.Println(err)
        return
    }

    // Print the ASCII string.
    fmt.Println("The ASCII representation of", hex, "is", string(ascii))
}

File: hex_to_ascii.js
function hexToASCII(hex) {
  var ascii = "";
  for (var i = 0; i < hex.length; i += 2) {
    ascii += String.fromCharCode(parseInt(hex.substring(i, i + 2), 16));
  }
  return ascii;
}

// Example
var hex = "48656c6c6f2c20576f726c6421";
var ascii = hexToASCII(hex);
console.log("The ASCII representation of " + hex + " is " + ascii);

